File: onlineChat/Hubs/ChatHub.cs
using Microsoft.AspNetCore.SignalR;
using onlineChat.Entity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace onlineChat.Hubs
{
    public class ChatHub : Hub
    {
        public async Task SendChat(Chat chat, string placeForChatId)
        {
            await Clients.All.SendAsync("SendChat", chat, placeForChatId);
        }
        public async Task SendMessage(string chatId, Message message, User author)
        {
            await Clients.Group(chatId).SendAsync("SendMessage", message, author);
        }
        public async Task JoinManyGroup(List<string> groupsId)
        {
            foreach (var groupId in groupsId)
                await Groups.AddToGroupAsync(Context.ConnectionId, groupId);

        }
        public async Task JoinGroup(string groupId)
        {
            await Groups.AddToGroupAsync(Context.ConnectionId, groupId);
        }
        public async Task JoinGroupMyFriend(string groupId,string connectionId)
        {
            await Groups.AddToGroupAsync(connectionId, groupId);
        }
        public override async Task OnConnectedAsync()
        {
            await Clients.Caller.SendAsync("GetConnetctedId", Context.ConnectionId);
            await base.OnConnectedAsync();
        }
    }
}

File: onlineChat/wwwroot/js/chatHub.js

class  ChatHub {
    static hubConnection;
    static async makeConnection(route) {
        ChatHub.hubConnection = new signalR.HubConnectionBuilder()
            .withUrl(route)
            .build();

       await ChatHub.hubConnection.start();
    }
    static setFunctionEventGetConnetctedId(functionEvent) {

        ChatHub.hubConnection.on("GetConnetctedId", functionEvent);
    }
    static invokeFunctionJoinGroup(groupId) {

        ChatHub.hubConnection.invoke("JoinGroup", groupId);
    }
    static invokeFunctionJoinGroupMyFriend(groupId,connectionId) {

        ChatHub.hubConnection.invoke("JoinGroupMyFriend", groupId, connectionId);
    }
    static invokeFunctionJoinManyGroup(groupsId) {

        ChatHub.hubConnection.invoke("JoinManyGroup", groupsId);
    }
    static invokeFunctionSendMessage(chatId,message,user) {

        ChatHub.hubConnection.invoke("SendMessage",chatId, message, user);
    }
    static setFunctionEvenSendMessage(functionEvent) {

        ChatHub.hubConnection.on("SendMessage", functionEvent);
    }
    static setFunctionEvenSendChat(functionEvent) {

        ChatHub.hubConnection.on("SendChat", functionEvent);
    }
    static invokeFunctionEvenSendChat(chat,placeForChatId) {

        ChatHub.hubConnection.invoke("SendChat", chat, placeForChatId);
    } 

}
